### PR TITLE
tests/ci: add tectonic-conformance-tests to jenkins code

### PR DESCRIPTION
**Pull request by cpanato**
cc @mxinden @Quentin-M

### CODE REVIEW

**Comment by mxinden** [comment]
Adding @rithujohn191 @squat here. As far as I know they have set up the conformance tests.

File: tests/jenkins-jobs/README.md [hunk 1]
```diff
@@ -44,6 +44,16 @@ Parameters:
 
 * `builder_image`: Tectonic-builder docker image with the upstream Terraform
 
+## Tectonic Conformance Tests
+
+This file (`tectonic_coformance_tests_job.groovy`) creates a Jenkins job called `tectonic-conformance-test` under `tectonic-tests` folder to run the conformance tests against the `Tectonic Installer` in the `master` branch or other specified branch.
```
Comment (squat) [inline tests/jenkins-jobs/README.md]: s/under/in the/

Comment (cpanato) [inline tests/jenkins-jobs/README.md]: done

File: tests/jenkins-jobs/README.md [hunk 2]
```diff
@@ -44,9 +44,19 @@ Parameters:
 
 * `builder_image`: Tectonic-builder docker image with the upstream Terraform
 
+## Tectonic Conformance Tests
+
+This file (`tectonic_coformance_tests_job.groovy`) creates a Jenkins job called `tectonic-conformance-test` in the `tectonic-tests` folder to run the conformance tests against the `Tectonic Installer` in the `master` branch or other specified branch.
```
Comment (squat) [inline tests/jenkins-jobs/README.md]: s/folder/directory/

Comment (squat) [inline tests/jenkins-jobs/README.md]: no need for backticks around "Tectonic Installer"

Comment (squat) [inline tests/jenkins-jobs/README.md]: s/`master` branch or other//

File: tests/jenkins-jobs/README.md [hunk 3]
```diff
@@ -44,9 +44,19 @@ Parameters:
...
+This job will run everyday around 12AM UTC time.
```
Comment (squat) [inline tests/jenkins-jobs/README.md]: s/around/at/

We are configuring this test and we should be able to say confidently when it runs.

File: tests/jenkins-jobs/tectonic_coformance_tests_job.groovy [hunk 4]
```diff
@@ -0,0 +1,64 @@
+#!/bin/env groovy​
+
+folder("tectonic-tests")
+
+job("tectonic-tests/tectonic-conformance-test") {
+  description("This job runs the Conformance tests for the tectonic-installer.\nThis job is manage by tectonic-installer.\nChanges here will be reverted automatically")
```
Comment (squat) [inline tests/jenkins-jobs/tectonic_coformance_tests_job.groovy]: If I change this file in Git, the changes will be reverted? What do you mean by "here"?

File: tests/jenkins-jobs/tectonic_coformance_tests_job.groovy [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+  logRotator(30, 100)
+  parameters {
+      stringParam('BRANCH', 'master', 'Which brach to run the conformance tests.')
```
Comment (squat) [inline tests/jenkins-jobs/tectonic_coformance_tests_job.groovy]: typo: s/brach/branch/

File: tests/jenkins-jobs/README.md (hunk 2 above)
Comment (cpanato) [inline tests/jenkins-jobs/README.md]: done

File: tests/jenkins-jobs/README.md (hunk 3 above)
Comment (cpanato) [inline tests/jenkins-jobs/README.md]: done

File: tests/jenkins-jobs/tectonic_coformance_tests_job.groovy (hunk 5 above)
Comment (cpanato) [inline tests/jenkins-jobs/tectonic_coformance_tests_job.groovy]: done

File: tests/jenkins-jobs/tectonic_coformance_tests_job.groovy (hunk 4 above)
Comment (cpanato) [inline tests/jenkins-jobs/tectonic_coformance_tests_job.groovy]: changes in the jenkins, when you read this in the jenkins, here then means in jenkins not in the git

**Comment by cpanato** [comment]
Can I merge @Quentin-M @rithujohn191 @mxinden ?

**Pull request merged**